File: appjava/src/main/java/com/byoutline/kickmaterial/model/ProjectIdAndSignature.java
package com.byoutline.kickmaterial.model;

import com.google.auto.value.AutoValue;

import java.util.Map;

/**
 * @author Sebastian Kacprzak <sebastian.kacprzak at byoutline.com>
 */
@AutoValue
public abstract class ProjectIdAndSignature {
    public static ProjectIdAndSignature create(int id, Map<String, String> queryParams) {
        return new AutoValue_ProjectIdAndSignature(id, queryParams);
    }

    public abstract int id();

    public abstract Map<String, String> queryParams();
}

File: app/src/main/java/com/byoutline/kickmaterial/model/ProjectIdAndSignature.kt
package com.byoutline.kickmaterial.model

import com.google.auto.value.AutoValue

/**
 * @author Sebastian Kacprzak <sebastian.kacprzak at byoutline.com>
 */
@AutoValue
abstract class ProjectIdAndSignature {

    abstract fun id(): Int

    abstract fun queryParams(): Map<String, String>

    companion object {
        fun create(id: Int, queryParams: Map<String, String>): ProjectIdAndSignature {
            return AutoValue_ProjectIdAndSignature(id, queryParams)
        }
    }
}
